### PR TITLE
Add Trace Analytics Docs

### PR DESCRIPTION
**Pull request by tylerbenson**
Includes updates to the Java APM with integration names to be used for config.

### Motivation
New capabilities for APM tracers

### Preview link
https://docs-staging.datadoghq.com/tyler/java-docs/tracing/advanced_usage/?tab=java
https://docs-staging.datadoghq.com/tyler/java-docs/tracing/languages/java/#integrations

### Additional Notes
Please wait to merge until released.

CC: @andrewsouthard1 @palazzem

### CODE REVIEW

File: content/tracing/advanced_usage/_index.md [hunk 1]
```diff
@@ -3,6 +3,83 @@ title: Advanced Languages instrumentation
...
+import io.opentracing.util.GlobalTracer;
+
+class SomeClass {
+  @Trace
+  void someMethod() {
+    final Span span = GlobalTracer.get().activeSpan();
+    // Span provided by @Trace annotation.
+    if (span != null) {
+      span.setTag(DDTags.SERVICE_NAME, "my-custom-service");
+      span.setTag(DDTags.EVENT_SAMPLE_RATE, 1.0);
```
Comment (palazzem) [inline content/tracing/advanced_usage/_index.md]: it should be `ANALYTICS_SAMPLE_RATE_KEY`

File: content/tracing/advanced_usage/_index.md [hunk 2]
```diff
@@ -3,6 +3,83 @@ title: Advanced Languages instrumentation
 kind: documentation
 ---
 
+## Trace Analytics Configuration
+
+Trace clients can send unsampled trace data to Trace Analytics for querying. Apply the following config to your application to enable this feature.
```
Comment (palazzem) [inline content/tracing/advanced_usage/_index.md]: @andrewsouthard1 I think in this overview we should explain a bit more what Trace Analytics does?

Comment (andrewsouthard1) [reply content/tracing/advanced_usage/_index.md]: Yeah, let's give a recap of what Trace Analytics is here.

>[Trace Analytics](https://docs.datadoghq.com/tracing/visualization/search/) is used to filter APM Data by [user-defined tags](https://docs.datadoghq.com/tracing/advanced_usage/?tab=java#custom-tagging) such as `customer_id`, `error_type` or `app_name` to help troubleshoot and filter your requests. Apply the following config to your application to enable this feature.

File: content/tracing/advanced_usage/_index.md (hunk 1 above)
Comment (tylerbenson) [reply content/tracing/advanced_usage/_index.md]: Should we deprecate the old tag or just remove it?

File: content/tracing/advanced_usage/_index.md (hunk 2 above)
Comment (tylerbenson) [reply content/tracing/advanced_usage/_index.md]: Done.

File: content/tracing/advanced_usage/_index.md [hunk 3]
```diff
@@ -3,6 +3,83 @@ title: Advanced Languages instrumentation
...
+      span.setTag(DDTags.SERVICE_NAME, "my-custom-service");
+      span.setTag(DDTags.ANALYTICS_SAMPLE_RATE, 1.0);
+    }
+  }
+}
+```
+
+{{% /tab %}}
+{{% tab "Python" %}}
+
```
Comment (palazzem) [inline content/tracing/advanced_usage/_index.md]: I would say to add a message in all missing tabs, such as "Coming Soon. Reach out to the Datadog support team to be part of the beta."

@andrewsouthard1 thoughts?

File: content/en/tracing/advanced_usage/_index.md [hunk 4]
```diff
@@ -3,6 +3,131 @@ title: Advanced Languages instrumentation
...
+    // Span provided by @Trace annotation.
+    if (span != null) {
+      span.setTag(DDTags.SERVICE_NAME, "my-custom-service");
+      span.setTag(DDTags.ANALYTICS_KEY, True);
+    }
+  }
+}
+```
+
+**Configure Sample Rate of APM Events**
```
Comment (majorgreys) [inline content/en/tracing/advanced_usage/_index.md]: Should this section be instead pulled out of the language-specific tabs?

File: content/en/tracing/advanced_usage/_index.md [hunk 5]
```diff
@@ -3,6 +3,131 @@ title: Advanced Languages instrumentation
...
+
+When enabling Trace Search & Analytics, the default sample rate is to collect 100% of APM Events. For any services that have been enabled for Trace Search & Analytics in the Tracing Client, you can adjusted the sampling rate for APM Events in your [APM settings][2].
+
+{{< img src="tracing/trace_sampling_ui.png" alt="Trace Sampling UI" responsive="true" style="width:100%;">}}
```
Comment (majorgreys) [inline content/en/tracing/advanced_usage/_index.md]: This image seems to replicate the intent of the above image.

**Comment by delner** [comment]
Ruby PR to merge into this one: https://github.com/DataDog/documentation/pull/4234

File: content/en/tracing/advanced_usage/_index.md (hunk 5 above)
Comment (andrewsouthard1) [reply content/en/tracing/advanced_usage/_index.md]: Thanks for pulling that out, that's a mistake. I will adjust before merge.

File: content/tracing/advanced_usage/_index.md (hunk 1 above)
Comment (rochdev) [reply content/tracing/advanced_usage/_index.md]: Didn't we settle on `ANALYTICS` or `ANALYTICS_KEY`?

Comment (palazzem) [reply content/tracing/advanced_usage/_index.md]: @rochdev this comment is outdated, check the latest: https://github.com/DataDog/documentation/pull/3989/files#diff-359edbc19bd14bae1400a15bd35a2fadR64 (`ANALYTICS_KEY`)

**Comment by andrewsouthard1** [comment]
👍

**Pull request merged**